third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -301,9 +306,12 @@ bool dds_device::impl::init()
                 {
                     LOG_DEBUG( "... device-options: " << j["n-options"] << " options received" );
 
-                    for( auto & option : j["options"] )
+                    for( auto & option_json : j["options"] )
                     {
-                        _options.push_back( dds_option::from_json( option, _info.name ) );
+                        auto option = dds_option::from_json( option_json, _info.name );
+                        _options.push_back( option );
+                        if( option->get_name() == "metadata_enabled" )
+                            _md_supported = option->get_value();
```

Comment:
This might make it `false` when it was previously `true`?

Reply:
No. Initialized to false, and set only here (when receiving device-options). Only one option can be named "metadata_enabled" and can change the value to `true` or leave it as `false`.